pkg/auth/zcapld/service.go

```diff
@@ -0,0 +1,117 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package zcapld
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/google/uuid"
+	cryptoapi "github.com/hyperledger/aries-framework-go/pkg/crypto"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/suite"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/suite/ed25519signature2018"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/util/signature"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	ariesstorage "github.com/hyperledger/aries-framework-go/pkg/storage"
+	"github.com/hyperledger/aries-framework-go/pkg/vdr/fingerprint"
+	"github.com/trustbloc/edge-core/pkg/zcapld"
+)
+
+const (
+	storeName = "zcap-capability"
+)
+
+// Service to provide zcapld functionality
+type Service struct {
+	keyManager kms.KeyManager
+	crypto     cryptoapi.Crypto
+	store      ariesstorage.Store
+}
+
+// New return zcap service
+func New(keyManager kms.KeyManager, crypto cryptoapi.Crypto, storeProv ariesstorage.Provider) (*Service, error) {
+	store, err := storeProv.OpenStore(storeName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open store %s: %w", storeName, err)
+	}
+
+	return &Service{keyManager: keyManager, crypto: crypto, store: store}, nil
+}
+
+// Create zcap payload
+func (s *Service) Create() ([]byte, error) {
+	rootCapability, err := s.createRootCapability()
```

Comment:
I should have made this clearer: there should be one root capability per resource.

So if a user creates Vault ID `12345` then the service needs to create a root capability _for that resource_ and store it. Then it should create a new zcap pointing to `12345` as the parent and with allowed actions.

Reply:
The invocation target of both the root capability and the child capability should identify the resource.

Reply:
I'm creating one root capability per Vault ID

Reply:
How are they specifying the vault ID as the invocation target? I don't see it.

Reply:
@llorllale the flow you explained it's there. Create method will be called from create Vault ID logic.

Reply:
Done